From: Hannes Reinecke <hare@suse.de>
Date: Fri, 18 Aug 2017 08:12:25 +0200
Subject: [PATCH] nvme-fc: kABI fix for defer_rcv() callback
Patch-Mainline: never, kABI fix for SLE12 SP3
References: bsc#1052384

Patch 0018-nvmet_fc-add-defer_req-callback-for-deferment-of-cmd.patch
added a new callback 'defer_rcv()' to the nvmet_fc_target_template
structure.
The patch reshuffles that new callback to the end so as not to break
kABI.

Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 include/linux/nvme-fc-driver.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/include/linux/nvme-fc-driver.h b/include/linux/nvme-fc-driver.h
index 1680227..70cff97 100644
--- a/include/linux/nvme-fc-driver.h
+++ b/include/linux/nvme-fc-driver.h
@@ -874,8 +874,6 @@ struct nvmet_fc_target_template {
 				struct nvmefc_tgt_fcp_req *fcpreq);
 	void (*fcp_req_release)(struct nvmet_fc_target_port *tgtport,
 				struct nvmefc_tgt_fcp_req *fcpreq);
-	void (*defer_rcv)(struct nvmet_fc_target_port *tgtport,
-				struct nvmefc_tgt_fcp_req *fcpreq);
 
 	u32	max_hw_queues;
 	u16	max_sgl_segments;
@@ -889,6 +887,10 @@ struct nvmet_fc_target_template {
 	u32	target_features;
 
 	u32	target_priv_sz;
+#ifndef __GENKSYMS__
+	void (*defer_rcv)(struct nvmet_fc_target_port *tgtport,
+				struct nvmefc_tgt_fcp_req *fcpreq);
+#endif
 };
 
 
-- 
1.8.5.6

